### PR TITLE
feat(17494): add generate-beta-commit

### PR DESCRIPTION
**Pull request by DDDDDanica**
## Explanation
This MR will address part of automation tasks in https://github.com/MetaMask/metamask-extension/issues/17493.

In order to trigger auto ci prod build for beta, we will first generate a script `yarn generate-beta-commit` to modify the current `package.json` version from `10.20.1` to `10.20.1-beta.0` under RC branch for further commit detection. 
Furthermore, if the dev is intended to bump the beta version, it is simple as running the same script `yarn generate-beta-commit` again to bump the version to `10.20.1-beta.1`.

One more thing to pay attention to, is such script will also assist in pushing the commits directly to your branch. Hence after the script you don't have any local changes you need to worry about.

The following snippet demonstrates the simple usage of this new script.

https://user-images.githubusercontent.com/12678455/217789678-85ea05e0-ee9a-4492-9845-7f825ae5005a.mov

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/signature-request-original/signature-request-original.component.js [hunk 1]
```diff
@@ -177,8 +177,12 @@ export default class SignatureRequestOriginal extends Component {
                 className="request-signature__row"
                 key={`request-signature-row-${index}`}
               >
-                <div className="request-signature__row-title">{`${name}:`}</div>
-                <div className="request-signature__row-value">{value}</div>
+                <div className="request-signature__row-title">
+                  {sanitizeString(`${name}:`)}
```
Comment (pedronfigueiredo) [inline ui/components/app/signature-request-original/signature-request-original.component.js:181]: Are any of these component changes related to the task? If not, can you extract them into another PR?

File: development/generate-beta-commit.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+async function start() {
+  let betaVersion;
+  if (VERSION.includes('beta')) {
+    const splitVersion = VERSION.split('-beta.');
+    const currentBetaVersion = Number(splitVersion[1]) + 1;
+    betaVersion = `${splitVersion[0]}-beta.${currentBetaVersion}`;
+    // Remove auto generated stableVersion an , after packageManager line to achieve bump
+    // You can find the issue here: https://github.com/yarnpkg/berry/issues/4328
+    await exec(
```
Comment (pedronfigueiredo) [inline development/generate-beta-commit.js:15]: The syntax of the sed commands and options can vary slightly between different platforms. I think `-i`, particularly, is not compatible with Windows.

Can you consider reading `package.json`, performing the operation on the in-memory JSON object, and then writing `package.json` using node standard packages? This would ensure cross-platform compatibility.

File: ui/components/app/signature-request-original/signature-request-original.component.js (hunk 1 above)
Comment (DDDDDanica) [reply ui/components/app/signature-request-original/signature-request-original.component.js]: They are the rebase legacies, removed 🙏

File: development/generate-beta-commit.js (hunk 2 above)
Comment (DDDDDanica) [reply development/generate-beta-commit.js:15]: there's some permission problem i encountered when reading / modifying package.json file.

So i was reading the solution in this issue, https://github.com/yarnpkg/berry/issues/4328, 
and using the script to manipulate the package.json directly,
and i got permission denied. 
```
#!/bin/bash

# Remove `stableVersion` before releasing, as it's buggy.
# https://github.com/yarnpkg/berry/issues/3868
echo "$(jq 'del(.stableVersion)' package.json)" >package.json
```

what if i use `awk` instead to perform?

Comment (DDDDDanica) [reply development/generate-beta-commit.js]: got some help in person and use fs module now . it is more clear now, thank you !

File: .gitignore [hunk 3]
```diff
@@ -65,4 +65,4 @@ test-results/
 !.yarn/plugins
 !.yarn/releases
 !.yarn/sdks
-!.yarn/versions
+.yarn/versions
```
Comment (pedronfigueiredo) [inline .gitignore:68]: For my understanding, can you explain this change?

Comment (DDDDDanica) [reply .gitignore:68]: When we bump version in this way there'll be fine generated locally under `.yarn/versions/` and each file is empty. That's why i ignored this folder. any other suggestions?

Comment (pedronfigueiredo) [reply .gitignore:68]: According to the [the docs](https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored), we might want to commit these files with the repo. Can you confirm this with @Gudahtt?

Comment (DDDDDanica) [reply .gitignore]: thanks for sharing the docs, was not aware of that config. now reverted back 🙏

**Review by pedronfigueiredo — APPROVED** [review]

**Review by danjm — APPROVED** [review]
this looks good to me!
